server.go

```diff
@@ -23,16 +53,126 @@ func NewServer(log *logrus.Logger, config *Config) (*Server, error) {
 		return &Server{}, err
 	}
 
+	var client *trace.Client
+	if config.StatsdAddress == "" {
+		client = dummyTraceClient()
+	} else {
+		client, err = trace.NewClient(config.StatsdAddress)
+		if err != nil {
+			return &Server{}, err
+		}
+	}
+
 	workers := make([]*Worker, config.WorkerCount)
 	for i := range workers {
-		workers[i] = NewWorker(log, config.WorkerSpanDepth, config.WorkerWatchDepth, expiration)
+		workers[i] = NewWorker(log, client, config.WorkerSpanDepth, config.WorkerWatchDepth, expiration)
 		go workers[i].Work()
 	}
-	return &Server{
+
+	// TODO make interval configurable
+	flushInterval := 10 * time.Second
+	ewmaAge := float64(flushInterval / time.Second)
+
+	s := &Server{
 		log:         log,
 		workerCount: int64(config.WorkerCount),
 		workers:     workers,
-	}, nil
+		traceClient: client,
```

Comment:
Make sure you https://godoc.org/github.com/stripe/veneur/trace#SetDefaultClient in main.go to set the default trace client! (Also, ensure that you set `trace.Service` to the right value, otherwise these spans will get dropped)

Reply:
(you'll probably have to make the field public to do that, but I think worth it)

Reply:
Why would it have to be public? good info on `trace.Service`, though - i guess that's how i can avoid having the `falconer` explicit prefix on everything 😄

Reply:
So the way veneur does it is:

1. `NewServer` sets the TraceClient field on the server it returns
2. `main` uses that trace client created by NewServer to set the global trace client.

This shouldn't affect anything that falconer itself does, but I find it good style to ensure everything that doesn't get a client passed through still reports to the address in the config.

Reply:
eeek. this global state is... 😭 😭 😭

Reply:
I could say the same thing about the `http` package, my dude (:

Reply:
Should be addressed now. i avoided exporting the field in favor of explicitly requiring one in the function sig.